ui/app/js/controllers/rule_new_controller.js

```diff
@@ -25,10 +25,20 @@ function($scope, $http, $modalInstance, CSRF, Releases, Rules, rules, rule, pr_c
   $scope.saveChanges = function () {
     $scope.saving = true;
     $scope.errors = {};
+    $scope.rate_error = {};
     CSRF.getToken()
     .then(function(csrf_token) {
       rule = angular.copy($scope.rule);
-      // rule.priority = '' + rule.priority;
+      
+      // Evaluate the values entered for priority and background rate.
+      $scope.integer_validation_errors = Helpers.integerValidator({'priority': rule.priority, 'rate': rule.backgroundRate});
```

Comment:
It would be ideal if you could integrate this into $scope.errors instead of introducing another variable for it. See below for an example of a bug that this causes...

Reply:
The reason I decided to define a separate variable for the UI errors(specifically the two in hand) was the `$scope.errors` is populated from the response after making the POST request.  And this is when backend errors are caught and displayed in the 2nd-to-last `<div>`. 

This bug fix specifies to validate the two fields before the POST request.

With the backend hacks back in place, even the *backgroundRate* and *Priority* error messages are displayed in this div.
<img width="815" alt="screen shot 2017-10-04 at 11 26 07" src="https://user-images.githubusercontent.com/28535137/31169718-346422b0-a902-11e7-81d9-8e7bac33b677.png">
<img width="1113" alt="screen shot 2017-10-04 at 11 23 04" src="https://user-images.githubusercontent.com/28535137/31169719-34672dca-a902-11e7-89c6-f4adf0403026.png">

And I have actually confirmed that we can use the `$scope.errors` in the meantime, to hold the UI validation errors, so making the change.

Reply:
Having looked further, I believe the format of the response -that in turn populates the $scope.error is not what was intended to be: Currently we have a response of the following format:
 `{detail: "Invalid input for backgroundRate. Not an integer. backgroundRate: u'y' is not a 'backgroundRate'", status: 400, title: "Bad Request", type: "about:blank"}`. I want to format the response to the format:
`{detail: {field1:field1-error, field2:field2-error...}, title:"Bad Request", type:""...}`. The rest of the fields will easily be styled from the **errors.detail.*** without much modifications to current codebase. Currently looking at where/how the response is composed so I can make the necessary changes.